weather_app_functions.R

```diff
@@ -0,0 +1,234 @@
+  
+  my_weather <- get_weather(location, apiKey)
+  
+  mymap <- image_scale(image_read(path = "App-1/Rplot.png"), "x400")
+  imageName <- paste("App-1/www/",my_weather$current$weather$icon,".png", sep = "")
+  icon <- image_scale(image_read("App-1/www/02d.png"), "x100")
+  
+  img <- c(mymap, icon)
+  myMapImage <- image_mosaic(img)
+  
+  image1 <- image_composite(mymap, icon, offset = "+20+20")
+  image2 <-image_composite(image1, icon, offset = "+280+180")
+  
+  image_write(myMapImage, path = "my_weather.png", format = "png")
+  image_write(image2, path = "my_weather2.png", format = "png")
+  
+  return(myMapImage)
+  
+}
+
+## Example for get_icon_map()
+get_icon_map("Amsterdam, Niederlande", sys.getenv("MY_API"))
+
+
+# This function takes a location (string) and an apiKey (string) as input and returns an image that corresponds 
+# to the current weather forecast for this location using openweathermap.org
+
+get_weather_image <- function(location, apiKey) {
+  
+  latitude <- tidygeocoder::geo_osm(location)$lat[1]
+  longitude <- tidygeocoder::geo_osm(location)$long[1]
+  
+  my_weather <- get_weather(location, apiKey)
+  
+  if (my_weather$current$weather$icon == "01d") {
+    
+    weather_image <- image_read("App-1/www/clearSky.jpg")
+    image_info(weather_image)
+    weather_image <- image_crop(weather_image, "1920x1280")
+    
+  } else if (my_weather$current$weather$icon == "02d" | my_weather$current$weather$icon == "03d" | my_weather$current$weather$icon == "04d") {
+    
+    weather_image <- image_read("App-1/www/clouds.jpg")
+    weather_image <- image_crop(weather_image, "1920x1280")
+    
+  } else if (my_weather$current$weather$icon == "09d" | my_weather$current$weather$icon == "10d") {
+    
+    weather_image <- image_read("App-1/www/rain.jpg")
+    
+  } else if (my_weather$current$weather$icon == "11d") {
+    
+    weather_image <- image_read("App-1/www/thunder2.jpg")
+    
+  } else if (my_weather$current$weather$icon == "13d") {
+    
+    weather_image <- image_read("App-1/www/snow.jpg")
+    
+  } else if (my_weather$current$weather$icon == "50d") {
+    
+    weather_image <- image_read("App-1/www/misty.jpg")
+    
+  }
+    
+  return(weather_image) 
+
+}
+
+## Example for get_weather_image()
+get_weather_image("Amsterdam, Niederlande", sys.getenv("MY_API"))
+
+
+# write a function that returns a GIF with weather icons that fly over the map of the users chosen location 
+# TO BE CONTINUED 
+
+get_weather_gif <- function(location, apiKey) {
+  
+  my_weather <- get_weather(location, apiKey)
+  
+  ## create a map and overlay it with the weather icon for the current weather
+  
+  # get the map and scale the png file
+  map <- get_map(location)
+  map <- image_scale(map, "x500")
+  
+  # get the icon and scale it
+  icon <- image_read(paste("App-1/www/", my_weather$current$weather$icon,".png", sep=""))
+  icon <- image_scale(icon, "x200")
+  
+  # create different png files that will make up the GIF

```

Comment:
This is really cool! I feel like the program (and this section in particular) will make it both useful and interesting to the user -- and goes above what you would get from more common weather apps.